dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/arithmetic/BeamSqlArithmeticExpression.java

```diff
@@ -29,14 +30,53 @@
  * Base class for all arithmetic operators.
  */
 public abstract class BeamSqlArithmeticExpression extends BeamSqlExpression {
-  private BeamSqlArithmeticExpression(List<BeamSqlExpression> operands, SqlTypeName outputType) {
+  private static final List<SqlTypeName> ORDERED_APPROX_TYPES = new ArrayList<>();
+  static {
+    ORDERED_APPROX_TYPES.add(SqlTypeName.TINYINT);
+    ORDERED_APPROX_TYPES.add(SqlTypeName.SMALLINT);
+    ORDERED_APPROX_TYPES.add(SqlTypeName.INTEGER);
+    ORDERED_APPROX_TYPES.add(SqlTypeName.BIGINT);
+    ORDERED_APPROX_TYPES.add(SqlTypeName.FLOAT);
+    ORDERED_APPROX_TYPES.add(SqlTypeName.DOUBLE);
+    ORDERED_APPROX_TYPES.add(SqlTypeName.DECIMAL);
+  }
+
+  protected BeamSqlArithmeticExpression(List<BeamSqlExpression> operands) {
+    super(operands, deduceOutputType(operands.get(0).getOutputType(),
```

Comment:
`BeamSqlArithmeticExpression(List<BeamSqlExpression> operands)` is used for `+`, `-` etc, so they don't have to specify the `outputType`:

```java
  public BeamSqlMinusExpression(List<BeamSqlExpression> operands) {
    super(operands);
  }
```

`BeamSqlArithmeticExpression(List<BeamSqlExpression> operands, SqlTypeName outputType)` is needed by `mod`, since it's `outputType` deduce logic is a little different from other arithmetic operators:

```java
  public BeamSqlModExpression(List<BeamSqlExpression> operands) {
    super(operands, operands.get(1).getOutputType());
  }
```